assets/js/modules/analytics/datastore/settings.test.js

```diff
@@ -1041,5 +1041,155 @@ describe( 'modules/analytics settings', () => {
 				).toBe( true );
 			} );
 		} );
+
+		describe( 'shouldPromptGA4DashboardView', () => {
+			it( 'should return false when the `ga4Reporting` feature flag is not enabled', () => {
+				// Delete the `ga4Reporting` feature flag if it exists.
+				enabledFeatures.delete( 'ga4Reporting' );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS )
+						.shouldPromptGA4DashboardView()
+				).toBe( false );
+			} );
+
+			it( 'should return false when the analytics-4 module is not active', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: false,
+						connected: false,
+					},
+				] );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS )
+						.shouldPromptGA4DashboardView()
+				).toBe( false );
+			} );
+
+			it( 'should return false when the analytics-4 module is not connected', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: false,
+					},
+				] );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS )
+						.shouldPromptGA4DashboardView()
+				).toBe( false );
+			} );
+
+			it( 'should return undefined when `isGA4DashboardView` is not loaded', async () => {
+				freezeFetch(
+					new RegExp(
+						'^/google-site-kit/v1/modules/analytics/data/settings'
+					)
+				);
+				enabledFeatures.add( 'ga4Reporting' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				expect(
+					registry.select( MODULES_ANALYTICS ).isGA4DashboardView()
+				).toBeUndefined();
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS )
+						.shouldPromptGA4DashboardView()
+				).toBeUndefined();
+
+				// Wait for resolvers to run.
+				await waitForDefaultTimeouts();
+			} );
+
+			it( 'should return false when `isGA4DashboardView` is true', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					dashboardView: 'google-analytics-4',
+				} );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS )
+						.shouldPromptGA4DashboardView()
+				).toBe( false );
+			} );
+
+			it( 'should return false when analytics-4 is gathering data', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+				provideModules( registry, [
+					{
+						slug: 'analytics-4',
+						active: true,
+						connected: true,
+					},
+				] );
+
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					dashboardView: 'universal-analytics',
```

Comment:
Again, once #6723 is merged we can use the `DASHBOARD_VIEW_UA4` constant here for consistency.

```suggestion
					dashboardView: DASHBOARD_VIEW_UA,
```